streams/src/main/java/org/apache/kafka/streams/state/internals/ValueAndTimestampSerializer.java

```diff
@@ -34,6 +34,25 @@
         timestampSerializer = new LongSerializer();
     }
 
+    public static boolean maskTimestampAndCompareValues(final byte[] left, final byte[] right) {
+        // adapted from Arrays.equals
+        if (left == right)
+            return true;
+        if (left == null || right == null)
+            return false;
+
+        final int length = left.length;
+        if (right.length != length)
+            return false;
+
+        // skip the timestamp when comparing just the values
+        for (int i = Long.BYTES; i < length; i++)
+            if (left[i] != right[i])
+                return false;
+
+        return true;
```

Comment:
prop: Instead of adding the comment, factor out this code to a method named `skipTimestampAndCompareValues()`.